internal/common/common.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func WaitForPodToStart(ctx context.Context, pods clientv1.PodInterface, objs *client.ObjectList) error {
```

Comment:
Should we convert this to a more generic function which waits for a pod to enter a phase.